pkg/logs/internal/tailers/file/tailer_windows.go

```diff
@@ -79,8 +79,16 @@ func (t *Tailer) readAvailable() (int, error) {
 		if n == 0 || err != nil {
 			return bytes, err
 		}
-		t.decoder.InputChan <- decoder.NewInput(inBuf[:n])
-		t.incrementLastReadOffset(n)
+		// try to put this buffer into the decoder channel
+		select {
+		case t.decoder.InputChan <- decoder.NewInput(inBuf[:n]):
+			t.incrementLastReadOffset(n)
+		default:
+			// the buffer is full, so pretend that this is all of the data that
+			// is available in the file at this point, in order to close the
+			// file and wait for the next poll interval.
+			return bytes, io.EOF
```

Comment:
Should you return `0` byes here? This will cause our [recorded bytes to go up](https://github.com/DataDog/datadog-agent/blob/main/pkg/logs/internal/tailers/file/tailer.go#L224) even though the bytes were thrown away. 

Also bytes is non `0` the sleep wont get hit in the tailer and we will go right back to this codepath immediately.